language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -119,22 +120,27 @@ private static Hover getObjectHoverMarkupContent(Documentation documentation, Ob
             params.add(header(3, ContextConstants.FIELD_TITLE) + CommonUtil.MD_LINE_SEPARATOR);
 
             params.addAll(classSymbol.fieldDescriptors().entrySet().stream()
-                    .map(fieldEntry -> {
+                    .filter(fieldEntry -> paramsMap.containsKey(fieldEntry.getKey())
+                            && !fieldEntry.getValue().qualifiers().contains(Qualifier.PRIVATE)).map(fieldEntry -> {
```

Comment:
I think we should show only the `public` fields. Because there can be fields with no qualifier which accounts to package-private I guess (the spec doesn't say that explicitly: https://ballerina.io/ballerina-spec/spec.html#object-field @pubudu91 @nadeeshaan can you confirm?). We have to check that. Ideally we should package private entries when the opened file is from the same package. We can do that later.

Reply:
That depends on the context. According to the context, the fields' visibility is changed. But in this case, I'm +1 to go with only the public fields, in order to maintain the consistancy.

Reply:
+1. Let me check on this

Reply:
Done